contracts/payoutStrategy/DirectPayoutStrategy/DirectPayoutStrategyImplementation.sol

```diff
@@ -0,0 +1,246 @@
+      "not round operator"
+    );
+    _;
+  }
+
+  function initialize(
+    address _alloSettings,
+    address _vaultAddress,
+    uint32  _roundFeePercentage,
+    address _roundFeeAddress
+  ) external initializer {
+    alloSettings = AlloSettings(_alloSettings);
+    vaultAddress = _vaultAddress;
+    roundFeePercentage = _roundFeePercentage;
+    roundFeeAddress = payable(_roundFeeAddress);
+  }
+
+  // --- Core methods ---
+
+  // @notice Update round fee percentage (only by ROUND_OPERATOR_ROLE)
+  /// @param _newFeePercentage new fee percentage
+  function updateRoundFeePercentage(uint32 _newFeePercentage) external isRoundOperator roundHasNotEnded {
+    roundFeePercentage = _newFeePercentage;
+    emit RoundFeePercentageUpdated(roundFeePercentage);
+  }
+
+  // @notice Update round fee address (only by ROUND_OPERATOR_ROLE)
+  /// @param _newFeeAddress new fee address
+  function updateRoundFeeAddress(address payable _newFeeAddress) external isRoundOperator roundHasNotEnded {
+    roundFeeAddress = _newFeeAddress;
+    emit RoundFeeAddressUpdated(roundFeeAddress);
+  }
+
+  /**
+   * @notice Invoked by a round operator to make signal that a pending application turns to IN REVIEW status.*
+   *
+   * @param _applicationIndex Application index
+   */
+  function setApplicationInReview(uint256 _applicationIndex) external isRoundOperator roundHasNotEnded {
+    if (!_isPendingRoundApplication(_applicationIndex)) revert DirectStrategy__setApplicationInReview_applicationInWrongStatus();
+    _inReviewApplications[_applicationIndex] = true;
+
+    emit ApplicationInReview(_applicationIndex, msg.sender);
+  }
+
+  /**
+   * @notice Invoked by a round operator to make direct payments of funds to a project application.
+   *
+   * @dev It can be used to pay from a given address using `ERC20.transferFrom`, or from
+   * the configured vault in which case the AllowanceModule should be set as a Safe Module on the Safe Multisig vault,
+   * and the caller as delegate on the AllowanceModule.
+   * Using `transferFrom` only allow to pay with ERC20 tokens, and requires the indicated vault previously approved this
+   * contract to use such ERC20 token on it behalf.
+   * This 2 options are handled by the `payment.vault` parameter, if it set to an address different from address(0) then
+   * the function will follow the `transferFrom` path.
+   *
+   * To complete the payment it is required for the project application to be on status ACCEPTED.
+   *
+   * @param _payment payment data
+   */
+  function payout(Payment calldata _payment) external nonReentrant isRoundOperator {
+    uint256 currentStatus = RoundImplementation(roundAddress).getApplicationStatus(_payment.applicationIndex);
+    if (currentStatus != uint256(ApplicationStatus.ACCEPTED)) revert DirectStrategy__payout_ApplicationNotAccepted();
+
+    address usedVault = vaultAddress;
+
+    // use transfer from
+    if (_payment.vault != address(0)) {
```

Comment:
I'm unclear on the naming here.
If the vault is set -> then you'd ERC20 from that vault to user
Else -> use the default vaultAddress 

In the if condition, _payment.value essentially is an EOA?

Reply:
if `_payment.vault` it is set, then it uses that address as the origin of funds to pay the grantee. It should work whether it is an EOA or smart contract, as long as this address previously approved the ERC20 token to be used by the strategy